providers/shared/fqdn_test.go

```diff
@@ -0,0 +1,72 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package shared
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFQDN(t *testing.T) {
+	tests := map[string]struct {
+		osHostname   string
+		expectedFQDN string
+		expectedErr  error
+	}{
+		"long_real_hostname": {
+			osHostname:   "elastic.co",
+			expectedFQDN: "elastic.co",
```

Comment:
Will temporary DNS errors cause this test to fail even if the logic is correct?

Reply:
Yeah, it will.  Just tested it with my Internet connection turned off.

I'm hoping such network/DNS unavailability errors are extremely rare, though.  Perhaps we could add a comment here and remove this test case if it starts proving to be flaky?

Reply:
I think a comment is a good idea, it won't necessarily be obvious that this requires an internet connection without context.

I'm fine waiting and seeing if this turns out to be flaky. Anyone testing this code will need to hit the real code path anyway so mocking it out doesn't necessarily help.